packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/ClaimButton/index.tsx

```diff
@@ -0,0 +1,195 @@
+"use client";
+
+import type { Chain } from "../../../../../../chains/types.js";
+import type { ThirdwebClient } from "../../../../../../client/client.js";
+import { getContract } from "../../../../../../contract/contract.js";
+import type { Hex } from "../../../../../../utils/encoding/helpers/is-hex.js";
+import type { TransactionButtonProps } from "../../../../../core/hooks/transaction/transaction-button-utils.js";
+import { useActiveAccount } from "../../../../hooks/wallets/useActiveAccount.js";
+import { TransactionButton } from "../../../TransactionButton/index.js";
+
+type ClaimParams =
+  | {
+      type: "ERC721";
+      quantity: bigint;
+    }
+  | {
+      type: "ERC1155";
+      quantity: bigint;
+      tokenId: bigint;
+    }
+  | ({
+      type: "ERC20";
+    } & ({ quantityInWei: bigint } | { quantity: string }));
+
+export type ClaimButtonProps = Omit<TransactionButtonProps, "transaction"> & {
+  contractAddress: Hex;
+  chain: Chain;
+  client: ThirdwebClient;
+  claimParams: ClaimParams;
+};
+
+/**
+ * This button is used to claim tokens (NFT or ERC20) from a given thirdweb Drop contract
+ * there are 3 type of Drop contract: NFT Drop (DropERC721), Edition Drop (DropERC1155) and Token Drop (DropERC20)
+ * Note: This button only works with thirdweb Drop contracts. For custom contract, please use TransactionButton
+ * @param props
+ * @returns A wrapper for TransactionButton
+ *
+ * @example
+ *
+ * ```tsx
+ * import { ClaimButton } from "thirdweb/react";
+ * import { ethereum } from "thirdweb/chains";
+ *
+ * // For NFT Drop (ERC721)
+ * <ClaimButton
+ *   contractAddress="0x..." // contract address of the NFT Drop
+ *   chain={ethereum}
+ *   client={client}
+ *   claimParams={{
+ *     type: "ERC721",
+ *     quantity: 1n, // claim 1 token
+ *   }}
+ * >
+ *   Claim now
+ * </ClaimButton>
+ *
+ *
+ * // For Edition Drop (ERC1155)
+ * <ClaimButton
+ *   contractAddress="0x..." // contract address of the Edition Drop
+ *   chain={ethereum}
+ *   client={client}
+ *   claimParams={{
+ *     type: "ERC1155",
+ *     quantity: 1n,
+ *     tokenId: 0n,
+ *   }}
+ * >
+ *   Claim now
+ * </ClaimButton>
+ *
+ *
+ * // For Token Drop (ERC20)
+ * <ClaimButton
+ *   contractAddress="0x..." // contract address of the Token Drop
+ *   chain={ethereum}
+ *   client={client}
+ *   claimParams={{
+ *     type: "ERC20",
+ *     quantity: "100", // claim 100 ERC20 tokens
+ *   }}
+ * >
+ *   Claim now
+ * </ClaimButton>
+ *
+ * ```
+ */
+export function ClaimButton(props: ClaimButtonProps) {
+  const { children, contractAddress, client, chain, claimParams } = props;
+  const contract = getContract({
+    address: contractAddress,
+    client,
+    chain,
+  });
+  const account = useActiveAccount();
+  return (
+    <TransactionButton
+      transaction={async () => {
+        switch (claimParams.type) {
+          case "ERC721": {
```

Comment:
for readability you could extract each of these CASEs into their own functions (can be defined inside the same file, but that way you can have clear split between logic and react for this